### PR TITLE
Avoid accidental EC2 recreation

### PR DESCRIPTION
**Pull request by cat5inthecradle**
We released a [change](https://console.aws.amazon.com/cloudformation/home?region=us-east-1#/stacks/changesets/changes?stackId=arn%3Aaws%3Acloudformation%3Aus-east-1%3A475661607190%3Astack%2Fstaging%2F45033800-919d-11e5-9d3a-5001dc3ed8d2&changeSetId=arn%3Aaws%3Acloudformation%3Aus-east-1%3A475661607190%3AchangeSet%2Fstaging-498c22fcfd8c9c14802e8f2b4d975594%2F13c6a8c9-6947-4a5d-9b39-204ac7e73b50) to the BlockDeviceMapping that triggered recreation of the EC2 instance. This is not good, as the Daemon server in staging, test, etc

## Links

- jira ticket: [INF-497](https://codedotorg.atlassian.net/browse/INF-497)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: aws/cloudformation/cloud_formation_stack.yml.erb [hunk 1]
```diff
@@ -432,6 +432,9 @@ Resources:
       Tags:
         - {Key: Name, Value: <%=frontends ? "#{environment}-daemon" : "!Ref 'AWS::StackName'"%>}
       BlockDeviceMappings:
+        # BlockDeviceMappings cannot be modified after instance creation
```
Comment (sureshc) [inline aws/cloudformation/cloud_formation_stack.yml.erb]: Should we be more explicit and state in the comment that modifying the Volume size will delete the daemon instance and recreate it?

**Comment by sureshc** [comment]
Should we set `DisableApiTermination` to `true` on the daemon EC2 Instance unless the environment == `adhoc`?

**Comment by sureshc** [comment]
Is there a mechanism we can use to take regular Snapshots of the daemon Volumes with Lifecycle management to prune the old Snapshots? AWS documentation indicates that it’s safest to stop an Instance before taking a Snapshot.

**Comment by cat5inthecradle** [comment]
> Is there a mechanism we can use to take regular Snapshots of the daemon Volumes with Lifecycle management to prune the old Snapshots? AWS documentation indicates that it’s safest to stop an Instance before taking a Snapshot.

Definitely safest to stop it - is there a window of time where we can afford to do that? Live snapshots are better than no snapshots though.

When I've done this in the past we implemented a general cloudwatch event and lambda(s) that looked for tags on EC2 instances that defined the backup rules (`backup: true, retain: 5` for example). But it looks like nowadays you can use [Data Lifecycle Manager](https://dev.to/paulmicheli/aws-automated-snapshot-cloudformation-1g6c).

File: aws/cloudformation/cloud_formation_stack.yml.erb [hunk 2]
```diff
@@ -469,6 +469,9 @@ Resources:
 <%  unless load_balancer -%>
           - !ImportValue VPC-ELBSecurityGroup # Accept HTTP traffic on daemon
 <%  end -%>
+<%  unless rack_env?(:adhoc) -%>
+    DisableApiTermination: true
```
Comment (sureshc) [inline aws/cloudformation/cloud_formation_stack.yml.erb]: Given that ERB and YAML nesting don’t interact well, a single line might be more readable. Something like:
```DisableApiTermination: <%= rack_env?(:adhoc) ? false : true %>```

**Review by sureshc — APPROVED** [review]

**Pull request merged**